compressor/management/commands/compress.py

```diff
@@ -208,47 +214,48 @@ def compress(self, log=None, **options):
         elif not isinstance(contexts, (list, tuple)):
             contexts = [contexts]
 
-        log.write("Compressing... ")
-        block_count = context_count = 0
-        results = []
-        offline_manifest = OrderedDict()
 
+        processes = options.get('processes', 1)
+        if processes > 1:
+            manager = multiprocessing.Manager()
+            offline_manifest_lock = manager.Lock()
+            offline_manifest = manager.dict()
+        else:
+            offline_manifest_lock = _DummyLock()
+            offline_manifest = {}
+        context_count = 0
+
+        global tasks
+        tasks = []
+        results = []
+        log.write("Compressing... ")
         for context_dict in contexts:
             context_count += 1
-            init_context = parser.get_init_context(context_dict)
-
             for template, nodes in compressor_nodes.items():
-                context = Context(init_context)
                 template._log = log
                 template._log_verbosity = verbosity
+                template_tasks = ((template, parser, node, context_dict,
+                        offline_manifest, offline_manifest_lock)
+                            for node in nodes)
+                if processes > 1:
+                    tasks += template_tasks
+                else:
+                    results += (_do_render(*task) for task in template_tasks)
```

Comment:
The single process special case is not necessary but AFAIU using process pool for single process would add unnecessary overhead - creating a new process and serializing/sending data between main process and pool process.

Also if there was any bug in multiprocess code this would allow a safe fallback.

(I will try to do some basic benchmarking during weekend)